cpp/ql/src/experimental/Security/CWE/CWE-266/IncorrectPrivilegeAssignment.ql

```diff
@@ -53,7 +53,13 @@ where
       fctmp.getTarget().hasGlobalOrStdName("fopen") or
       fctmp.getTarget().hasGlobalOrStdName("open")
     ) and
-    not fctmp.getArgument(1).getValue().matches("r%") and
+    (
+      fctmp.getArgument(1).getValue().matches("%a%") or
+      // unfortunately cannot use numeric value here because // O_APPEND is defined differently on different OSes:
+      // https://github.com/red/red/blob/92feb0c0d5f91e087ab35fface6906afbf99b603/runtime/definitions.reds#L477-L491
+      // this may introduce false negatives
+      fctmp.getArgument(1).getValueText().matches("%O_APPEND%")
```

Comment:
I don't think the `%`s in this match will do what you want because, say, `O_APPEND | O_WRONLY` does not have useful value text.  You might get better results with something like
```
fctmp.getArgument(1).getValueText().matches("O_APPEND") or
fctmp.getArgument(1).(BitwiseOrExpr).getAChild*().getValueText().matches("O_APPEND")
```

Reply:
Nice... Interesting techinque. On a side note, I wasn't right about `O_APPEND` the key flag is `O_CREAT`.

Reply:
Good table comparing the fopen and open flags in https://pubs.opengroup.org/onlinepubs/9699919799/functions/fopen.html